src/NHibernate/Action/CollectionAction.cs

```diff
@@ -79,6 +79,11 @@ protected internal ISessionImplementor Session
 
 		#region IExecutable Members
 
+		public string[] UpdateTimestampsSpaces
+		{
+			get { return Persister.OwnerEntityPersister.HasUpdateTimestampsCache ? Persister.CollectionSpaces : null; }
```

Comment:
I think that here we should check `ElementPersister` instead of `OwnerEntityPersister`. Is there a reason why used `OwnerEntityPersister`?

Reply:
How to get ElementPersister its not accessible do I need to cast Persister as AbstractCollectionPersister then I could get ElementPersister?

Reply:
My bad, here we should use `PropertySpaces` instead, but if we add the mentioned lookups to the session factory, this won't be needed anymore.